text/0510-dynamodb-global-table.md

```diff
@@ -0,0 +1,1179 @@
+});
+```
+
+You can also allocate capacities for your GSIs:
+
+* You only need to provide write and read capacity for GSIs where you want it to be different than capacity specified
+for the global table. If not specified, it uses the same value as that of the global table.
+
+  ```typescript
+  new GlobalTable(tableStack, 'GlobalTable', {
+    tableName: 'FooTable',
+    partitionKey: {
+      name: 'FooHashKey',
+      type: AttributeType.STRING,
+    },
+    sortKey: {
+      name: 'FooRangeKey',
+      type: AttributeType.STRING,
+    },
+    billingMode: BillingMode.PROVISIONED,
+    write: Capacity.autoscaled({ max: 70 }),
+    read: Capacity.fixed(40),
+    globalSecondaryIndex: [{
+      indexName: 'UniqueGsiName',
+      partitionKey: {
+        name: 'FooRangeKey',
+        type: AttributeType.STRING,
+      },
+      write: Capacity.autoscaled({ max: 90 }),
+      read: Capacity.autoscaled({ max: 60 }),
+    }],
+    replicas: [{
+      region: 'us-east-1',
+    }],
+  });
+  ```
+
+* You can provide read capacity for GSIs where you want it to be different than the capacity specified for the replica.
+
+  ```typescript
+  new GlobalTable(tableStack, 'GlobalTable', {
+    tableName: 'FooTable',
+    partitionKey: {
+      name: 'FooHashKey',
+      type: AttributeType.STRING,
+    },
+    sortKey: {
+      name: 'FooRangeKey',
+      type: AttributeType.STRING,
+    },
+    billingMode: BillingMode.PROVISIONED,
+    write: Capacity.autoscaled({ max: 70 }),
+    globalSecondaryIndex: [{
+      indexName: 'UniqueGsiName',
+      partitionKey: {
+        name: 'FooRangeKey',
+        type: AttributeType.STRING,
+      },
+    }],
+    replicas: [{
+      region: 'us-east-1',
+      read: Capacity.autoscaled({ max: 70 }),
+      globalSecondaryIndexOptions: [
+        {
+          indexName: 'UniqueGsiName',
+          read: Capacity.fixed(55),
+        },
+      ],
+    }],
+  });
+  ```
+
+__NOTE:__
+
+* You can create up to 20 global secondary indexes.
+* You can only create or delete one global secondary index in a single stack operation.
+
+#### Local Secondary Indexes
+
+You can add local secondary indexes to your global table. These will be the same for each replica of
+the table.
+
+```typescript
+const globalTable = new GlobalTable(stack, 'FooTable', {
+  partitionKey: { name: 'Foo', type: AttributeType.STRING },
+  sortKey: { name: 'Bar', type: AttributeType.STRING },
+  localSecondaryIndex: [
+    {
+      indexName: 'FooTableLsi',
+      sortKey: { name: 'Foo', type: AttributeType.STRING },
+    },
+  ],
+});
+```
+
+You can add local secondary index with `addLocalSecondaryIndex` method.
+
+```typescript
+globalTable.addLocalSecondaryIndex({
+  indexName: 'FooTableLsi',
+  sortKey: { name: 'Foo', type: AttributeType.STRING },
+});
+```
+
+__NOTE:__
+
+* You need a sort key to define a local secondary index.
+* You can create up to five local secondary indexes.
+
+#### Encryption
```

Comment:
Again this could be an enum-like class: `

```ts
encryption: TableEncryption.regionKeyArns({
  'us-east-1': 'arn:kms:...',
  'us-east-2': 'arn:kms:...',
});
```

Reply:
Taking into account that the table itself also needs a key, and that should probably be an `IKey` for single-region consistency:

```ts
encryption: TableEncryption.customerManagedKey(tableKey, {
  replicaKeyArns: {
    'us-east-1': 'arn:kms:...',
    'us-east-2': 'arn:kms:...',
  }
});
```

Perhaps?